source/solvers/heat_transfer_assemblers.cc

```diff
@@ -56,10 +56,52 @@ HeatTransferAssemblerCore<dim>::assemble_matrix(
 
       const auto velocity = scratch_data.velocity_values[q];
 
-      // Calculation of the magnitude of the velocity for the
-      // stabilization parameter
+      // Calculation of the magnitude of the velocity for the stabilization
+      // parameter
       const double u_mag = std::max(velocity.norm(), 1e-12);
 
+      // Temperature gradient information for DCDD stabilization
+      Tensor<1, dim> dcdd_temperature_gradient;
+      if (is_steady(method))
+        {
+          dcdd_temperature_gradient = scratch_data.temperature_gradients[q];
+        }
+      else
+        {
+          dcdd_temperature_gradient =
+            scratch_data.previous_temperature_gradients[0][q];
+        }
```

Comment:
I think this could be more readable if done in a single line
```suggestion
      Tensor<1, dim> dcdd_temperature_gradient = is_steady(method) ?  scratch_data.temperature_gradients[q] : scratch_data.previous_temperature_gradients[0][q] ; 
```